
# ------------------------------ Config parsing------------------------- #
configfile: "config/config.yaml"
LENGTH = str(config['SEQTK']['length']) 

# ------------------------------ Include Snakefiles ------------------------- #
include: "rules/common.smk"
# include: "rules/0.1_viromeqc_beforeqc.smk"
# include: "rules/0.2_fastp.smk"
# include: "rules/0.3_remove_host_reads.smk"
# include: "rules/0.4_viromeqc_afterqc.smk"
# include: "rules/1.1_penguin.smk"
# include: "rules/1.2_length_filtering.smk"
include: "rules/2.1_concatenate_all_sample_fastas"
# include: "rules/2.3_metaquast.smk"
# include: "rules/2.2_checkv.smk"
# include:  "rules/2.3_genomad.smk"
# include: "rules/9_salmon_afterqc_reads_to_jaeger_virus_contigs.smk"
# ------------------------------ Targets ------------------------------------ #
samples_df = samplesheet_to_df(config['samplesheet'])
SAMPLES = samples_df['sample_id'].values.tolist()

rule all:
    input:
        # FASTP output
        # expand("results/{sample}/fastp/{sample}.fastp.html", sample = SAMPLES),
        # VIROMEQC_BEFOREQC output
        # expand("results/{sample}/viromeqc_beforeqc/{sample}.viromeqc_beforeqc.txt", sample = SAMPLES),
        # SAMTOOLS remove host reads output
        # expand("results/{sample}/remove_host_reads/{sample}_1.remove_host_reads.fastq.gz", sample = SAMPLES),
        # expand("results/{sample}/remove_host_reads/{sample}_2.remove_host_reads.fastq.gz", sample = SAMPLES),
        # VIROMEQC_AFTERQC output
        # expand("results/{sample}/viromeqc_afterqc/{sample}.viromeqc_afterqc.txt", sample = SAMPLES),
        # PENGUIN output
        # expand("results/{sample}/penguin/{sample}.scaffolds.fasta", sample = SAMPLES),
        # LENGTH FILTERING output
        # expand("results/{sample}/penguin/{sample}.scaffolds_gt{length}.fasta", sample = SAMPLES, length=LENGTH),
        # CONCATENATE FASTA FILES output
        "combined_results/scaffolds/penguin_all.scaffolds.fasta",
        # METAQUAST output
        # expand("results/{sample}/penguin_metaquast/report.txt", sample = SAMPLES),
        # JAEGER output
        # expand("results/{sample}/jaeger/{sample}.scaffolds_gt" + LENGTH + "_default_jaeger.tsv", sample = SAMPLES, length=LENGTH),
        # JAEGER virus extract output
        # expand("results/{sample}/jaeger/{sample}.jaeger_virus_gt10000_ids.txt", sample = SAMPLES),
        # expand("results/{sample}/jaeger/{sample}.jaeger_virus_gt10000_seqs.fasta", sample = SAMPLES),
        # VCLUST output
        # expand("results/{sample}/vclust/{sample}.vclust_fltr.txt", sample = SAMPLES),
        # expand("results/{sample}/vclust/{sample}.vclust_ani.txt", sample = SAMPLES),
        # expand("results/{sample}/vclust/{sample}.vclust_species.txt", sample = SAMPLES),
        # expand("results/{sample}/vclust/{sample}.vclust_genera.txt", sample = SAMPLES),
        # CHECKV output
        # expand("results/{sample}/checkv/quality_summary.tsv", sample = SAMPLES),
        # # GENOMAD output
        # expand("results/{sample}/genomad/{sample}.scaffolds_gt" + LENGTH + "_summary.log", sample = SAMPLES, length=LENGTH),
        # # SALMON output
        # expand("results/{sample}/salmon_jaeger/", sample = SAMPLES),
